### PR TITLE
LPS-176902 There should be a default service account for each virtual instance

### PR DESCRIPTION
**Pull request by pei-jung**

### CODE REVIEW

**Comment by pei-jung** [comment]
ci:test:sf

**Comment by pei-jung** [comment]
ci:test:relevant

**Comment by pei-jung** [comment]
ci:test:usm

File: modules/apps/users-admin/users-admin-impl/src/main/java/com/liferay/users/admin/internal/search/UserBulkReindexer.java [hunk 1]
```diff
@@ -49,7 +49,7 @@ public void reindex(long companyId, Collection<Long> classPKs) {
 		indexableActionableDynamicQuery.setCompanyId(companyId);
 		indexableActionableDynamicQuery.setPerformActionMethod(
 			(User user) -> {
-				if (!user.isDefaultUser()) {
+				if (!(user.isDefaultUser() && !user.isServiceAccountUser())) {
```
Comment (drewbrokke) [inline modules/apps/users-admin/users-admin-impl/src/main/java/com/liferay/users/admin/internal/search/UserBulkReindexer.java:52]: Would it be possible to use `isRegularUser` instead?

File: modules/apps/users-admin/users-admin-impl/src/main/java/com/liferay/users/admin/internal/search/spi/model/index/contributor/UserModelIndexerWriterContributor.java [hunk 2]
```diff
@@ -42,7 +42,7 @@ public void customize(
 
 		batchIndexingActionable.setPerformActionMethod(
 			(User user) -> {
-				if (!user.isDefaultUser()) {
+				if (!(user.isDefaultUser() && !user.isServiceAccountUser())) {
```
Comment (drewbrokke) [inline modules/apps/users-admin/users-admin-impl/src/main/java/com/liferay/users/admin/internal/search/spi/model/index/contributor/UserModelIndexerWriterContributor.java:45]: Same question as in `UserBulkReindexer`.

**Review by drewbrokke — COMMENTED** [review]
Just a couple of questions, otherwise this looks good, @pei-jung!

**Comment by pei-jung** [comment]
ci:test:sf

**Comment by pei-jung** [comment]
ci:test:relevant

**Comment by pei-jung** [comment]
ci:test:sf

**Comment by pei-jung** [comment]
ci:test:relevant

**Comment by pei-jung** [comment]
ci:test:usm

**Comment by pei-jung** [comment]
ci:test:sf

**Comment by pei-jung** [comment]
ci:test:relevant

**Comment by pei-jung** [comment]
@patriciajeanperez do you mind checking the failures and see if they are related? Thank you!

**Comment by patriciajeanperez** [comment]
@pei-jung no more related failures. I added the label

**Comment by pei-jung** [comment]
ci:forward:force

**Comment by pei-jung** [comment]
ci:forward:force

**Comment by pei-jung** [comment]
ci:stop

**Comment by pei-jung** [comment]
Manually forwarded to https://github.com/brianchandotcom/liferay-portal/pull/131423.

**Pull request closed without merging**